crates/storage/provider/src/providers/state/historical.rs

```diff
@@ -201,9 +230,9 @@ impl<'a, 'b, TX: DbTx<'a>> StateRootProvider for HistoricalStateProviderRef<'a,
 impl<'a, 'b, TX: DbTx<'a>> StateProvider for HistoricalStateProviderRef<'a, 'b, TX> {
     /// Get storage.
     fn storage(&self, address: Address, storage_key: StorageKey) -> Result<Option<StorageValue>> {
-        match self.storage_history_lookup(address, storage_key)? {
-            HistoryInfo::NotYetWritten => Ok(None),
-            HistoryInfo::InChangeset(changeset_block_number) => Ok(Some(
+        Ok(match self.storage_history_lookup(address, storage_key)? {
```

Comment:
ill frontrun @mattsse  nit and say to not use such a big Ok wrapper, and return it individually on each branch